tests/src/org.testcontainers/testcontainers/1.19.8/build.gradle

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    implementation('org.reflections:reflections:0.10.2')
+    implementation "org.testcontainers:testcontainers:$libraryVersion"
+    runtimeOnly "org.slf4j:slf4j-simple:1.7.36"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    binaries {
+        test {
+            // We have to manually add the metadata for JNA, as testcontainers use JNA
+            configurationFileDirectories.from(tck.metadataRoot.dir('net.java.dev.jna/jna/5.8.0'))
```

Comment:
Have you tried to add JNA metadata in `META-INF/native-image` directory of the test? Check [this PR](https://github.com/oracle/graalvm-reachability-metadata/pull/257/files#diff-9957db8c55035c5fd697f91c7a7a682f694549386d2fb4117ddd42f995e0af95) for example. That way your tests will have additional metadata, but only for their run. Metadata provided like this won't be part of the metadata we are shipping in the release, and will be used only for test purposes.

More examples of such a usages can be found in:
https://github.com/oracle/graalvm-reachability-metadata/pull/118
https://github.com/oracle/graalvm-reachability-metadata/pull/141
https://github.com/oracle/graalvm-reachability-metadata/pull/397

Reply:
Copied the `build.gradle` as-is from from Testcontainers tests 1.17.6.

However it seems to works without JNA here so I removed the entry.